python/ray/workflow/step_executor.py

```diff
@@ -110,11 +111,18 @@ def execute_workflow(workflow: "Workflow") -> "WorkflowExecutionResult":
     # Stage 1: prepare inputs
     workflow_data = workflow.data
     inputs = workflow_data.inputs
+    workflow_outputs = []
     with workflow_context.fork_workflow_step_context(
             outer_most_step_id=None, last_step_of_workflow=False):
-        workflow_outputs = [
-            execute_workflow(w).persisted_output for w in inputs.workflows
-        ]
+        for w in inputs.workflows:
+            static_ref = w.ref
+            if static_ref is None:
+                # The input workflow is not a reference to an executed
+                # workflow .
+                output = execute_workflow(w).persisted_output
+                static_ref = WorkflowStaticRef(step_id=w.step_id, ref=output)
```

Comment:
I'm actually quite confused about `WorkflowStaticRef`. I'm thinking that when to use WorkflowStaticRef and when to use other things. Are we modeling every output into this? Should we name it `WorkflowOutput` ?

We probably want to do some subtraction here. THink about this, we have a lot of types for different things, can they be unified into a common model?